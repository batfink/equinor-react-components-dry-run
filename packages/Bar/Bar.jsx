import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Bar = ({ className, ...props }) => <Wrapper className={ className } { ...props } />

const Wrapper = styled.div`
    padding: 20px;
    background: brown;
    position: relative;
    font: 12px sans-serif;

    &::before {
        position: absolute;
        content: 'Bar';
        top: 2px;
        left: 2px;
        padding: 0.2em;
        background: rgba(255,255,255,0.8);
    }
`

Bar.propTypes = {
    className: PropTypes.string,
}

Bar.defaultProps = {
    className: '',
}

export default Bar
