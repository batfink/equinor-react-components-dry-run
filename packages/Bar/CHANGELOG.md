# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

<a name="0.0.3"></a>
## [0.0.3](https://github.com/batfink/equinor-react-components-dry-run/compare/@equinor-internal/bar@0.0.2...@equinor-internal/bar@0.0.3) (2018-09-11)

**Note:** Version bump only for package @equinor-internal/bar
