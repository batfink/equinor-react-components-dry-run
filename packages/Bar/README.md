Bar
====

Usage
-----

```bash
yarn add @equinor-internal/bar
```
