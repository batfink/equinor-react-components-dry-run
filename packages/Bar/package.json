{
    "name": "@equinor-internal/bar",
    "main": "Bar",
    "version": "0.0.3",
    "author": "Victor Nystad",
    "license": "UNLICENSED",
    "dependencies": {
        "prop-types": "^15.6.2",
        "react": "^16.5.0",
        "styled-components": "^3.4.5"
    }
}
