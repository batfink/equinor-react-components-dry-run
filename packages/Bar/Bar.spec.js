import React from 'react'
import Bar from './Bar'
import Enzyme from 'enzyme'
import 'jest-styled-components'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

const { mount } = Enzyme

describe('Bar', () => {
    it('should render same as snapshot', () => {
        const tree = mount(<Bar active={ false } />)

        expect(tree).toMatchSnapshot()
        expect(tree).toHaveStyleRule('color', 'gray')
    })
})
