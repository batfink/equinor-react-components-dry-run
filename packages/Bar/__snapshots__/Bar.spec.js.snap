// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bar should render same as snapshot 1`] = `
.c0 {
  padding: 20px;
  position: relative;
  color: gray;
  font: 12px sans-serif;
}

.c0::before {
  position: absolute;
  content: 'Bar';
  top: 2px;
  left: 2px;
  padding: 0.2em;
  background: rgba(255,255,255,0.8);
}

<Bar
  active={false}
  className=""
>
  <Bar__Wrapper
    active={false}
    className=""
  >
    <div
      className="c0"
    />
  </Bar__Wrapper>
</Bar>
`;
